profiler/src/ProfilerEngine/Datadog.Profiler.Native/StackSamplerLoop.cpp

```diff
@@ -52,7 +56,8 @@ StackSamplerLoop::StackSamplerLoop(
     IStackSnapshotsBufferManager* pStackSnapshotsBufferManager,
     IManagedThreadList* pManagedThreadList,
     ISymbolsResolver* pSymbolResolver,
-    IWallTimeCollector* pWallTimeCollector
+    IWallTimeCollector* pWallTimeCollector,
```

Comment:
Should we remove `IWallTimeCollector` or add a `ICpuTimeCollector`? We could also define `typedef ICollector<WallTimeSampleRaw> IWallTimeCollector;` in `ICollector.hpp` for a shortcut

Reply:
I left both IWallTimeCollector and ICollector<RawCpuSample> to show the previous and new way to implement providers. The next commit will remove IWallTimeCollector. 
I tend to prefer explicit ICollector<RawxxxSample> because it makes it clear what is the input collected instances